### PR TITLE
Node status in Popup // Option page

### LINKED ISSUE #9: Allow lookup service address to be customizable

### PR DESCRIPTION
**Pull request by ngotchac**
Closes #9

Better Account Card in the Pop-Up + Ground for node connection status

Created an Option page where one can:

1. Enable/Disable the Identity Extraction/Augmentation
2. Enable/Disable the Web3 Integration
3. Modify the node and dapps URL
4. Modify the Lookup-Service URL (as a fallback)

~Need #48 to be merged before being able to finish this one.~

Some options are hidden by default, but available via an `Advanced Options` checkbox

### CODE REVIEW

File: popup/component/app.js [hunk 1]
```diff
@@ -58,35 +65,65 @@ export default class App extends Component {
 
   render () {
     const { store } = this.props;
-    const { enabled, extractions } = this.state;
+    const { augmentationEnabled, extractions, status, url } = this.state;
 
     return (
       <div className={ styles.container }>
         <div className={ styles.header }>
           <h1 className={ styles.title }>Web3 Injection</h1>
```
Comment (tomusdrw) [inline popup/component/app.js]: The title doesn't make much sense now

File: popup/component/app.js [hunk 2]
```diff
@@ -58,35 +65,65 @@ export default class App extends Component {
...
+      case 'disconnected':
+      default:
+        phrase = 'Not connected to';
+        break;
+    }
+
+    return (
+      <div className={ styles.status }>
+        <span className={ iconClassName } />
+        <span>{ phrase } { url }</span>
```
Comment (tomusdrw) [inline popup/component/app.js]: I would just display "local node" instead of URL (we can put url to a tooltip/title)

File: popup/component/app.js (hunk 1 above)
Comment (tomusdrw) [reply popup/component/app.js]: When nothing was extracted on the website, I would rephrase this to "No identities have been found on the page."

File: popup/component/app.js (hunk 2 above)
Comment (tomusdrw) [reply popup/component/app.js]: Maybe it should also hint consequences when you are not connected to the node, like: "Web3 websites will not work.".

File: options/component/app.js [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+        <div className={ styles.option }>
+          <div className={ styles.checkbox }>
+            <CheckBox
+              checked={ showAdvanced }
+              className={ styles.check }
+              onChange={ this.handleToggleAdvanced }
+            />
+          </div>
+          <div>
+            Show Advanced Options
```
Comment (tomusdrw) [inline options/component/app.js]: Label is not clickable and it's a bit annoying.

File: options/component/app.js [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+        <div className={ styles.option }>
+          <div className={ styles.switch }>
+            <Switch
+              checked={ augmentationEnabled }
+              className={ styles.check }
+              onChange={ this.handleToggleAugmentation }
+            />
+          </div>
+          <div>
+            Identity Augmentation
```
Comment (tomusdrw) [inline options/component/app.js]: Label is not clickable and it's a bit annoying (same with Web3)

File: background/transport.js [hunk 5]
```diff
@@ -53,10 +70,49 @@ export default class Transport {
...
+          break;
+
+        case 'disconnected':
+        default:
+          ctx.fillStyle = 'rgba(231, 76, 60, 0.25)';
+          break;
+      }
+
+      ctx.strokeStyle = 'white';
+      ctx.beginPath();
```
Comment (tomusdrw) [inline background/transport.js]: Have you tried drawing it in upper left corner? I think now it looks a bit crowdy when badge is displayed as well.

File: background/lookup.js [hunk 6]
```diff
@@ -239,7 +252,7 @@ export default class Lookup {
         ? 'emailHash'
         : method;
 
-      return fetch(`https://id.parity.io:8443/?${lookupMethod}=${input}`)
+      return fetch(`${this.lookupURL}/?${lookupMethod}=${input}`.replace(/\/+/g, '/'))
```
Comment (tomusdrw) [inline background/lookup.js]: It's not realy future proof (imagine if `input` can be something more then email in future). I think it would be better to trim possible trailing `/` from `lookupURL` instead. Also it should be done when altering the options not with every request.

File: options/component/app.js [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+  }
+
+  renderAdvancedOptions () {
+    const { DAPPS, lookupURL, UI } = this.state;
+
+    return (
+      <div className={ styles.advanced }>
+        <div className={ [ styles.option, styles.optionInput ].join(' ') }>
+          <div className={ styles.input }>
+            <TextField
```
Comment (tomusdrw) [inline options/component/app.js]: I think it would be cool to perform a check if the service is accessible before saving the options.
If we cannot connect (or the output is invalid) we should display a warning (but not prevent from saving such options).

**Review by tomusdrw — COMMENTED** [review]
Couple of grumbles. But overall it looks awesome :+1:
Good job as always :)

File: background/transport.js (hunk 5 above)
Comment (ngotchac) [reply background/transport.js]: That works too, fixed

**Comment by ngotchac** [comment]
@tomusdrw Addressed PR Grumbles :)

**Comment by tomusdrw** [comment]
Seems that it also closes #52

File: popup/component/app.js [hunk 8]
```diff
@@ -93,7 +89,20 @@ export default class App extends Component {
...
 
-  renderStatus (status, url) {
+  renderHint (status) {
+    if (status === 'connected' || status === 'connecting') {
+      return null;
+    }
+
+    return (
+      <p className={ styles.error }>
+        You are not connected to a local Parity Node. Seemless integration
```
Comment (tomusdrw) [inline popup/component/app.js]: Seamless

File: popup/component/app.js [hunk 9]
```diff
@@ -121,7 +130,7 @@ export default class App extends Component {
     return (
       <div className={ styles.status }>
         <span className={ iconClassName } />
-        <span>{ phrase } { url }</span>
+        <span>{ phrase } local node</span>
```
Comment (tomusdrw) [inline popup/component/app.js]: Sorry to complicate this even more, but I think it would also be handy to have a network name displayed there instead, like:

"Connected to Ropsten"

**Review by tomusdrw — COMMENTED** [review]
A typo and a feature request :)

**Comment by tomusdrw** [comment]
Just did some tests:
1. When you open a page while not being connected to the node it displays identities correctly, but when you run the node it stops displaying identities:
screenshot from 2017-02-10 08-33-42

2. Red icon is not drawn right after installation:
screenshot from 2017-02-10 08-42-07

File: popup/component/app.js (hunk 9 above)
Comment (ngotchac) [reply popup/component/app.js]: You got it ! :+1:

**Comment by ngotchac** [comment]
@tomusdrw Was this the same chain ? By default, without node, it fetches the identities from the main net. If you run a node on the test net, there will be some cache issues I think :/

**Comment by tomusdrw** [comment]
I'm not sure now, it's possible it was different chain. Can't we re-compute identities?

**Comment by ngotchac** [comment]
Yes, this cache needs to be clear automatically on network/lookup URL change

**Comment by ngotchac** [comment]
@tomusdrw Okay, the cache should work better now (caching per retrieving method). The little connection icon should work better and the network name is displayed

**Review by tomusdrw — APPROVED** [review]
:+1:
